app/raid.js

```diff
@@ -1,225 +1,294 @@
-		if (!this.roles.valor) { this.roles.valor = member.guild.roles.find('name', 'Valor'); }
-		if (!this.roles.instinct) { this.roles.instinct = member.guild.roles.find('name', 'Instinct'); }
+  findRaid(channel, member, args) {
+    const raids = args
+      .map(arg => this.getRaid(channel, member, arg))
+      .filter(raid => {
+        return raid !== undefined;
+      });
 
-		// add extra data to "member"
-		member.additional_attendees = 0;
+    let raid;
+
+    if (raids.length > 0) {
+      raid = raids[0];
+    } else {
+      raid = this.getRaid(channel, member);
+    }
 
-		// add some extra raid data to remember
-		raid_data.id = id;
-		raid_data.timestamp = Date.now();
-		raid_data.attendees = [ member ];
+    const nonRaidArgs = args
+      .filter(arg => {
+        return this.getRaid(channel, member, arg) === undefined;
+      });
 
-		if (channel_raid_map) {
-			channel_raid_map.set(id, raid_data);
-		} else {
-			channel_raid_map = new Map();
-			channel_raid_map.set(id, raid_data);
-			this.raids.set(channel.id, channel_raid_map);
-		}
+    return {'raid': raid, 'args': nonRaidArgs};
+  }
 
-		this.raids_counter++;
+  getAttendeeCount(options) {
+    let attendees = [];
+    let length = 0;
 
-		return { raid: raid_data };
-	}
+    // get attendee data via given raid data, or map data in order to find the attendee data
+    if (options.raid) {
+      attendees = options.raid.attendees;
+    } else {
+      if (!options.channel || !options.member || isNaN(options.raid_id)) {
+        throw ('Need raid data in order to get attendee count.');
+      }
+      attendees = this.getRaid(options.channel, options.member, options.raid_id).attendees;
+    }
 
-	getRaid(channel, raid_id) {
-		return this.raids.get(channel.id).get(raid_id);
-	}
+    length = attendees.length;
 
-	getAttendeeCount(options) {
-		var attendees = [];
-		var length = 0;
+    for (let i = 0; i < attendees.length; i++) {
+      const attendee = attendees[i];
+      length += attendee.additional_attendees;
+    }
 
-		// get attendee data via given raid data, or map data in order to find the attendee data
-		if (options.raid) {
-			attendees = options.raid.attendees;
-		} else {
-			if (!options.channel || !options.member || isNaN(options.raid_id)) { throw ('Need raid data in order to get attendee count.'); }
-			attendees = this.getRaid(options.channel, options.raid_id).attendees;
-		}
+    return length;
+  }
 
-		length = attendees.length;
+  getMessage(channel, member, raid_id) {
+    return this.getRaid(channel, member, raid_id).message;
+  }
 
-		for (let i=0; i<attendees.length; i++) {
-			var attendee = attendees[i];
-			length += attendee.additional_attendees;
-		}
+  setMessage(channel, member, raid_id, message) {
+    this.getRaid(channel, member, raid_id).message = message;
+  }
 
-		return length;
-	}
+  addAttendee(channel, member, raid_id, additional_attendees = 0) {
+    const raid_data = this.getRaid(channel, member, raid_id);
+    let index;
```

Comment:
Curious, it looks like you changed a lot of the "index" variables to "const", why is this one "let"?